src/components/inputModal/index.tsx

```diff
@@ -1,62 +1,106 @@
 import { schema } from "utils/inputFormSchema";
 import { MODAL_TYPE } from "utils/enums";
-import InputTextField from "components/inputModal/InputTextField";
-import InputSelectField from "components/inputModal/InputSelect";
-import {grades, subjects} from "utils/index";
+import InputTextField from "components/inputModal/inputTextField";
+import InputSelectField from "components/inputModal/inputSelect";
+import { grades, subjects } from "utils/index";
+import { IStudentRaw } from "state/ducks/students/types";
+import { ActionType } from "typesafe-actions";
+import {
+  addStudent,
+  fetchStudents,
+  updateStudent,
+} from "state/ducks/students/actions";
 interface IModalProps {
   visible: boolean;
   setVisible: Function;
-  // isEdit: boolean;
+  fetchStudents: () => ActionType<typeof fetchStudents>;
+  addStudent: (payload: IStudentRaw) => ActionType<typeof addStudent>;
+  updateStudent: (payload: IStudentRaw) => ActionType<typeof updateStudent>;
+  studentData: IStudentRaw;
+  setStudentData: Function;
 }
 
 const StudentInputModal: React.FC<IModalProps> = (props) => {
-  const initState = {
+  const initState = props.studentData
+    ? { ...props.studentData }
+    : {
+        name: "",
+        marks: 0,
+        subject: "",
+        grade: "",
+      };
+
+  const [payload, setPayload] = useState<IStudentRaw>({
     name: "",
+    marks: 0,
     subject: "",
-    marks: "",
-    grades: "",
-  };
-
-  // const modalType = props?.isEdit ? MODAL_TYPE.EDIT : MODAL_TYPE.ADD;
-  const modalType = MODAL_TYPE.ADD;
+    grade: "",
+  });
+  const modalType = props?.studentData?._id ? MODAL_TYPE.EDIT : MODAL_TYPE.ADD;
 
   const {
-    register,
     handleSubmit,
     reset,
     control,
     formState: { errors, isSubmitSuccessful },
-  } = useForm({
+  } = useForm<IStudentRaw>({
     mode: "onTouched",
     reValidateMode: "onChange",
     resolver: yupResolver(schema),
-    defaultValues: initState,
   });
 
+  useEffect(() => {
+    if (modalType === "Add")
+      reset({
+        name: "",
+        marks: 0,
+        subject: "",
+        grade: "",
+      });
+  }, [modalType]);
+
+  useEffect(() => {
+    if (modalType === "Edit") {
+      reset(props.studentData);
+    }
+  }, [props.studentData]);
+
   useEffect(() => {
     if (isSubmitSuccessful) {
-      reset();
-      console.log("Reset form!");
+      reset(initState);
+      if (modalType === "Edit") {
+        props.updateStudent({ ...payload, _id: props?.studentData._id });
+        props.fetchStudents();
+      } else {
+        props.addStudent(payload);
+      }
+      props.setStudentData();
     }
-  }, [isSubmitSuccessful, reset, props.visible]);
+  }, [isSubmitSuccessful]);
+
+  const onSubmit = (values: IStudentRaw) => {
+    props.setVisible(false);
+    setPayload(values);
+  };
 
-  const onSubmit = (values: object) => {
-    console.log("Values:::", values);
+  const handleClose = () => {
     props.setVisible(false);
+    props.setStudentData();
```

Comment:
this line does nothing. if you want to set the `studentData` to null or empty object use `setStudentData(null)` or `setStudentData({})`. (reason why your modal was maintaining previous state 🙃)